src/test/keyboard-shortcut.service.spec.ts

```diff
@@ -1,127 +1,346 @@
 import { BrowserModule } from '@angular/platform-browser';
 import { async, TestBed } from '@angular/core/testing';
-import { } from 'jasmine';
+import {} from 'jasmine';
 
 import { KeyboardShortcutsService } from '../keyboard-shortcut.service';
-import { KeyboardKeys, KeyboardShortcutCombination } from '../libraries/listener.library';
+import {
+  KeyboardKeys,
+  KeyboardShortcutCombination,
+  IKeyboardShortcutListenerOptions
+} from '../libraries/listener.library';
+import { Component } from '@angular/core';
 
 describe('KeyboardShortcutService', () => {
+  let service: KeyboardShortcutsService;
 
-    let service: KeyboardShortcutsService;
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [],
+      imports: [BrowserModule],
+      providers: [KeyboardShortcutsService]
+    });
+    service = TestBed.get(KeyboardShortcutsService);
+  }));
+
+  // ================================================
 
-    beforeEach(async(() => {
-        TestBed.configureTestingModule({
-            declarations: [
-            ],
-            imports: [
-                BrowserModule,
-            ],
-            providers: [
-                KeyboardShortcutsService,
-            ],
+  describe(':: testing methods', () => {
+    describe(':: listen', () => {
+      it('should return an IListenerHandle', () => {
+        const answer = service.listen({
+          keyBinding: [],
+          handler: () => '',
+          description: ''
         });
-        service = TestBed.get(KeyboardShortcutsService);
-    }));
+        const isIListenerHandle = answer.remove ? true : false;
+        expect(isIListenerHandle).toBe(true);
+      });
 
-    // ================================================
+      it('should add a listener to the collection', () => {
+        const startingCount = service.listeners_read_only.length;
+        expect(startingCount).toBe(0);
+        service.listen({ keyBinding: [], handler: () => '', description: '' });
+        const endingCount = service.listeners_read_only.length;
+        expect(endingCount).toBe(1);
+      });
 
-    describe(':: testing methods', () => {
+      it('should warn if keyBinding is black listed', () => {
+        spyOn(console, 'warn');
+        service.listen({
+          keyBinding: [KeyboardKeys.Ctrl, 'c'],
+          handler: () => '',
+          description: ''
+        });
+        expect(console.warn).toHaveBeenCalled();
+      });
 
-        describe(':: listen', () => {
+      it('should warn if keyBinding is black listed', () => {
```

Comment:
this appears to be a duplicate test

Reply:
to be clear, I do see that it is not a duplicate, since the order of the keybinding is different, but we should discuss how that could be better handled